### PR TITLE
#3234 exclude extra field when represent model

### LINKED ISSUE #3234: when extra is allowed, representation is break
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
python -c "import pydantic.utils; print(pydantic.utils.version_info())"
             pydantic version: 1.8.2
            pydantic compiled: False
                 install path: /Users/Sol/Project/pydantic/pydantic
               python version: 3.9.5 (default, Jun 23 2021, 22:47:04)  [Clang 11.0.3 (clang-1103.0.32.62)]
                     platform: macOS-10.15.6-x86_64-i386-64bit
     optional deps. installed: ['devtools', 'dotenv', 'email-validator', 'typing-extensions']
```

```py
from pydantic import BaseModel
from pydantic import Extra



class Foo(BaseModel):
    a: int

    class Config:
        extra = Extra.allow


f = Foo(a=1, b=2)
breakpoint()
print(f) # << raise error. not only 'print', also str(Model), breakpoint().. etc

```

because

field b that allowed is not in Model.fields

in `main.BaseModel`

```
class BaseModel(Representation, metaclass=ModelMetaclass):

    ...

    def __repr_args__(self) -> 'ReprArgs':
        return [(k, v) for k, v in self.__dict__.items() if self.__fields__[k].field_info.repr] # KeyError
```

### PR DESCRIPTION
**Pull request by COCOLMAN**
## Change Summary

has been modified to exclude unpredictable "extra" fields from "__repr__"

## Related issue number

fix #3234

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by COCOLMAN** [comment]
please review

**Comment by COCOLMAN** [comment]
I thought it was right to exclude "extra field".

Let me know if I'm thinking in a different direction from main

File: changes/3234-cocolman.md [hunk 1]
```diff
@@ -0,0 +1 @@
+ exclude unpredictable "extra" fields from "repr"
```
Comment (samuelcolvin) [inline changes/3234-cocolman.md:1]:
```suggestion
Fix display of `extra` fields with model `__repr__`
```

File: pydantic/main.py [hunk 2]
```diff
@@ -853,7 +853,9 @@ def __eq__(self, other: Any) -> bool:
             return self.dict() == other
 
     def __repr_args__(self) -> 'ReprArgs':
-        return [(k, v) for k, v in self.__dict__.items() if self.__fields__[k].field_info.repr]
+        return [
+            (k, v) for k, v in self.__dict__.items() if self.__fields__.get(k) and self.__fields__[k].field_info.repr
```
Comment (samuelcolvin) [inline pydantic/main.py:857]:
```suggestion
            (k, v) for k, v in self.__dict__.items() if k not in self.__fields__ or self.__fields__[k].field_info.repr
```

We **should** included these properties in `repr()`

**Review by samuelcolvin — COMMENTED** [review]
thanks for this, good catch

**Comment by samuelcolvin** [comment]
please update.

**Comment by COCOLMAN** [comment]
updated

File: tests/test_main.py [hunk 3]
```diff
@@ -194,6 +194,16 @@ class Config:
...
 
 
+def test_allow_extra_repr():
+    class Model(BaseModel):
+        a: float = ...
+
+        class Config:
+            extra = Extra.allow
+
+    assert str(Model(a='10.2', b=12)) == f'a={10.2}'
```
Comment (samuelcolvin) [inline tests/test_main.py:204]:
```suggestion
    assert str(Model(a='10.2', b=12)) == 'a=10.2 b=12'
```

**Comment by samuelcolvin** [comment]
HI @COCOLMAN you seem to have prevented me from contributing to your PR which means I can't fix the small issue in tests.

Please accept this change. In future it would be easier if you gave permissions to maintainers to contribute to your pull requests for situations exactly like this.

**Comment by COCOLMAN** [comment]
Thx.  

I will check `Allow edits by maintainers` next time

**Comment by samuelcolvin** [comment]
Thanks so much.

**Pull request merged**